src/components/OuterModal/WaitForTxApproval.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useRef, useEffect, useLayoutEffect } from 'react'
+import styled from 'styled-components'
+
+import Modal from 'components/common/Modal'
+import { openGlobalModal, useGlobalModalContext, setOuterModalContext, closeGlobalModal } from './SingletonModal'
+
+const WaitForTxWrapper = styled.div`
+  font-size: 1.3em;
+`
+
+interface GlobaModalContextSlice {
+  pendingTxApprovals: Set<number | string>
+}
+
+const PenidngTxApprovalsMessage: React.FC<{ txNumber: number }> = ({ txNumber }) => {
+  return (
+    <>
+      <h4>No response from wallet for pending transaction</h4>
+      <p>There are currently {txNumber} transactions waiting for approval for your wallet.</p>
+      <p>
+        Mesa has detected a potential transaction timeout. Please validate with your wallet that you have properly
+        accepted or rejected the transaction.
+      </p>
+      <p>Do you need more time signing the transaction?</p>
+    </>
+  )
+}
+
+const NoTxsWillClose: React.FC = () => {
+  return (
+    <>
+      <h4>All transactions have been handled</h4>
+      <p>Mesa has received responses for all pending transactions.</p>
+      <p> This modal will close in a moment.</p>
+    </>
+  )
+}
+
+// time to display `Will Close` message
+const CLOSE_DELAY = 3000
+
+export const WaitForTxApprovalMessage: React.FC = () => {
+  const { pendingTxApprovals } = useGlobalModalContext<GlobaModalContextSlice>()
+
+  const [initialHeight, setInitialHeight] = useState(0)
+
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  // layout effect for sync rerendering on style change
+  // otherwise we get layout thrashing
+  useLayoutEffect(() => {
+    if (!wrapperRef.current) return
+    setInitialHeight(wrapperRef.current.offsetHeight)
+  }, [])
```

Comment:
should I ask?

Reply:
LayoutEffect is triggered synchronously before browser paint
with the usual Effect it'll `render first-> paint-> measure -> change style -> render -> paint`, you'll see a slight jump on style change
with Layout Effect `render first-> measure -> change style -> render -> paint`, we trigger rerender before browser paints